app.js

```diff
@@ -11,39 +11,99 @@ app.use(express.json());
 
 // Iteration 1 - Connect to MongoDB
 // DATABASE CONNECTION
+const mongoose = require("mongoose");
+const MONGODB_URI = "mongodb://127.0.0.1:27017/express-mongoose-recipes-dev";
 
-
+mongoose
+  .connect(MONGODB_URI)
+  .then((x) => console.log(`Connected to Mongo! Database name: "${x.connections[0].name}"`))
+  .catch((err) => console.error("Error connecting to mongo", err));
 
 // ROUTES
-//  GET  / route
+//  GET  / route - This is just an example route
 app.get('/', (req, res) => {
-    res.send("<h1>Lab | Express Mongoose Recipes</h1>");
+    res.send("<h1>LAB | Express Mongoose Recipes</h1>");
 });
 
 
-//  Iteration 4 - Create recipe route
+//  Iteration 3 - Create a Recipe route
 //  POST  /recipes route
-
-
-//  Iteration 5 - Read all recipes
+app.post("/recipes", (req, res) => {
+    Recipe.create({
+        title: req.body.title, 
+        instructions: req.body.instructions, 
+        level: req.body.level, 
+        ingredients: req.body.ingredients,
+        image: req.body.image, 
+        duration: req.body.duration, 
+        isArchived: req.body.isArchived, 
+        created: req.body.created,
+
+    })
+.then((createdRecipe)=>{
+    res.status(201).json(createdRecipe);
+})
+.catch((err)=>{
+    res.status(500).json({message: "Error while creating a new recipe"})
+})
+})
+
+//  Iteration 4 - Get All Recipes
 //  GET  /recipes route
-
-
-//  Iteration 6 - Read a single recipe
+app.get("/recipes", (req, res) =>{
+    Recipe.find()
+    .then((allRecipes) =>{
+        res.status(201).json(allRecipes)
+    })
+    .catch((err)=>{
+        res.status(500).json({message: "Error while getting all the recipes"})
+    })
+})
+
+
+//  Iteration 5 - Get a Single Recipe
 //  GET  /recipes/:id route
-
-
-//  Iteration 7 - Update a single recipe
+app.get("/recipes/:id", (req, res) =>{
+    Recipe.findById(req.params.id)
```

Comment:
also applys for the id in the params 
```suggestion
const { id } = req.params
Recipe.findById(id)
```